crates/cubecl-runtime/src/server.rs

```diff
@@ -41,9 +51,15 @@ where
     /// Given a resource as bytes, stores it and returns the memory handle.
     fn create(&mut self, data: &[u8]) -> Handle;
 
+    /// Given a resource as bytes with `shape`, stores it and returns the tensor handle.
+    fn create_tensor(&mut self, data: &[u8], shape: Vec<usize>, elem_size: usize) -> TensorHandle;
```

Comment:
For Burn it's going to be a very big pain to have more than just one kind of Handles, I would modify `Handle` instead of returning a new type.